src/mscorlib/src/System/Array.cs

```diff
@@ -1706,11 +1707,13 @@ public IEnumerator GetEnumerator()
                 throw new ArgumentNullException("keys");
             if (keys.Rank != 1 || (items != null && items.Rank != 1))
                 throw new RankException(Environment.GetResourceString("Rank_MultiDimNotSupported"));
-            if (items != null && keys.GetLowerBound(0) != items.GetLowerBound(0))
+            int keysLowerBound = keys.GetLowerBound(0);
+            int itemsLowerBound = items.GetLowerBound(0);
```

Comment:
The original code was special-casing `items == null` - I think you still need to do that. E.g.:
```
int lowerBound =  keys.GetLowerBound(0);
if (items != null && lowerBound != items.GetLowerBound(0)) 
...
 you can use lowerBound for both keys and items since you know they are equal...
```